pythia/utils/checkpoint.py

```diff
@@ -42,7 +42,7 @@ def __init__(self, trainer):
 
         self.models_foldername = os.path.join(self.ckpt_foldername, "models")
         if not os.path.exists(self.models_foldername):
-            os.makedirs(self.models_foldername)
+            os.makedirs(self.models_foldername, exist_ok=True)
```

Comment:
Can we not change this? This might have dangerous consequences where we overwrite trained models by mistake.

Reply:
This is fine. This is an issue in case of distributed training which is already fixed in our dev branch. If you see the `if` statement above you will understand. This happens when there is a race condition and one of the jobs have already made the folder, then the whole thing fails. That's why `exist_ok=True` needs. `exist_ok` doesn't overwrite, it just doesn't throw an error if the folder is already there.

Reply:
Hi, From Oleksii's experience, without this change, the program frequently crashes (over 50% time) in distributed training due to a race condition, since multiple processes are trying to make directories, and there's no lock on these lines.